vault/resource_ldap_secret_backend.go

```diff
@@ -0,0 +1,684 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/terraform-provider-vault/util"
+	"github.com/hashicorp/vault/api"
+	"log"
+)
+
+func ldapSecretBackendResource() *schema.Resource {
+	fields := map[string]*schema.Schema{
```

Comment:
fields pulled from https://github.com/hashicorp/vault-plugin-secrets-openldap/blob/0d09d4c5613321f936a42cafcfc9d9f5559a9e8c/path_config.go#L63

Reply:
Thanks for being so thorough here! Like I mentioned earlier, super confusing that the LDAP secrets engine returns config fields from LDAP auth.. 

I'm thinking we can start with what's documented at https://developer.hashicorp.com/vault/api-docs/secret/ldap#configuration-management? That is _probably_ the most accurate source for the config parameters used in LDAP secrets. If we missed something, we can always expand from there. Lmk your thoughts!

Reply:
I believe we can remove all of the following:

```
FieldCaseSensitiveNames
FieldMaxTTL
FieldTLSMaxVersion
FieldTLSMinVersion
FieldTTL
```

And this one is labeled deprecated in the docs so maybe we should keep it?

```
FieldLength
```